rest/model/conversions.go

```diff
@@ -27,50 +30,56 @@ type convertFuncs struct {
 	inverter  string
 }
 
+type convertFnOpts struct {
+	in       types.Type
+	outIsPtr bool
+	outType  string
+}
+
 // conversionFn uses the specific naming conventions of type conversion funcs in conversions.gotmpl
 // to find the correct generated type conversion function. It returns the name of the conversion function
 // and the name of the inverse conversion function
-func conversionFn(in types.Type, outIsPtr bool, generatedConversions map[string]string) (convertFuncs, error) {
-	if _, inIsPrimitive := in.(*types.Basic); inIsPrimitive {
-		return simplePtrs(in.String(), outIsPtr, generatedConversions)
+func conversionFn(opts convertFnOpts, generatedConversions map[typeInfo]string) (convertFuncs, error) {
+	if _, inIsPrimitive := opts.in.(*types.Basic); inIsPrimitive {
+		return simplePtrs(opts.in.String(), opts.outType, false, opts.outIsPtr, generatedConversions)
 	}
-	if intype, inIsPtr := in.(*types.Pointer); inIsPtr {
+	if intype, inIsPtr := opts.in.(*types.Pointer); inIsPtr {
 		value, isPrimitive := intype.Elem().(*types.Basic)
 		if isPrimitive {
-			return simplePtrs(value.String(), outIsPtr, generatedConversions)
+			return simplePtrs(value.String(), opts.outType, true, opts.outIsPtr, generatedConversions)
```

Comment:
This is getting kind of gnarly--maybe typeInfo can also include inIsPtr and outIsPtr so it can be used more broadly (like here).

Reply:
good point, changed to pass that struct around